python/hidet/graph/ops/definitions/activation.py

```diff
@@ -87,6 +88,131 @@ def __init__(self, x: Tensor):
+
+
+class CeluOp(UnaryElementwiseOp):
+    def __init__(self, x: Tensor, alpha: float = 1.0) -> Tensor:
+        super().__init__(
+            x,
+            op=lambda v: prim.max(x.dtype(0.0), v)
+            + prim.min(x.dtype(0.0), x.dtype(alpha) * (prim.exp(v / x.dtype(alpha)) - x.dtype(1.0))),
+            name='celu',
+        )
+
+
+class RReluOp(UnaryElementwiseOp):
+    def __init__(self, x: Tensor, lower: float = 1 / 8, upper: float = 1 / 3) -> Tensor:
+        super().__init__(
+            x, op=lambda v: if_then_else(v >= 0, v, prim.random(x.dtype(lower), x.dtype(upper)) * v), name='rrelu'
+        )
+
+
+class LogSigmoidOp(UnaryElementwiseOp):
+    def __init__(self, x: Tensor):
+        super().__init__(x, op=lambda v: prim.log(x.dtype(1.0) / (x.dtype(1.0) + prim.exp(-v))), name='logsigmoid')
+
+
+class HardShrinkOp(UnaryElementwiseOp):
+    def __init__(self, x: Tensor, lambda_val: float = 0.5):
+        super().__init__(
+            x,
+            op=lambda v: if_then_else(
+                v > x.dtype(lambda_val), v, if_then_else(v < x.dtype(-lambda_val), v, x.dtype(0.0))
+            ),
+            name='hardshrink',
+        )
+
+
+class TanhShrinkOp(UnaryElementwiseOp):
+    def __init__(self, x: Tensor):
+        super().__init__(
+            x, op=lambda v: v - (prim.exp(v) - prim.exp(-v)) / (prim.exp(v) + prim.exp(-v)), name='tanhshrink'
+        )
+
+
+class SoftSignOp(UnaryElementwiseOp):
+    def __init__(self, x: Tensor):
+        super().__init__(x, op=lambda v: v / (x.dtype(1.0) + prim.abs(v)), name='softsign')
+
+
+class SoftPlusOp(UnaryElementwiseOp):
+    def __init__(self, x: Tensor, beta: int = 1, threshold_val: int = 20):
+        super().__init__(
+            x,
+            op=lambda v: if_then_else(
+                v * x.dtype(beta) <= x.dtype(threshold_val),
+                (x.dtype(1.0) / x.dtype(beta)) * prim.log(x.dtype(1.0) + prim.exp(x.dtype(beta) * v)),
+                v,
+            ),
+            name='softplus',
+        )
+
+
+class SoftShrinkOp(UnaryElementwiseOp):
+    def __init__(self, x: Tensor, lambda_val: float = 0.5):
+        super().__init__(
+            x,
+            op=lambda v: if_then_else(
+                v > x.dtype(lambda_val),
+                v - x.dtype(lambda_val),
+                if_then_else(v < x.dtype(-lambda_val), v + x.dtype(lambda_val), x.dtype(0.0)),
+            ),
+            name='softshrink',
+        )
+
+
+class SoftmaxOp(Operator):
+    def __init__(self, x: Tensor, axis: int = 1):
+        axis = normalize_dim(axis, len(x.shape))
+        super().__init__(
+            inputs=[x], task=SoftmaxTask(input_like(x, 'x'), axis), attributes={'axis': axis}, name='softmax'
+        )
+
+
+class SoftminOp(Operator):
+    def __init__(self, x: Tensor, axis: int = 1):
+        axis = normalize_dim(axis, len(x.shape))
+        super().__init__(
+            inputs=[-x], task=SoftmaxTask(input_like(x, 'x'), axis), attributes={'axis': axis}, name='softmin'
```

Comment:
It is better to do the minus in the Task instead of Operator definition.

Reply:
It's better to move the softmax task definition to activitation.py as well.

Reply:
Another option to deal with the `-` in softmin is to do this in the `def softmin()` function.